Userland/Libraries/LibWeb/CSS/ComputedValues.h

```diff
@@ -88,6 +88,14 @@ struct BackgroundLayerData {
     CSS::LengthPercentage size_y { CSS::Length::make_auto() };
     CSS::Repeat repeat_x { CSS::Repeat::Repeat };
     CSS::Repeat repeat_y { CSS::Repeat::Repeat };
+
+    // Cache offsets to avoid recalculating each paint.
+    struct ResolvedOffset {
+        float resolved_basis = NAN;
+        float resolved_value = 0;
+    };
+    mutable ResolvedOffset resolved_offset_x;
+    mutable ResolvedOffset resolved_offset_y;
```

Comment:
This information belongs in the paint tree, not with the CSS computed values.

Reply:
Maybe I'll just remove this for now, since there's not clearly a good way to put it on the paintable (which we don't have access to here anyway). I think to move this to the paintable a decent chunk of the logic from the painting code would have to be moved out (which would be good, but not for this PR).